nilearn/_utils/data_gen.py

```diff
@@ -967,15 +963,27 @@ def create_fake_bids_dataset(
     return bids_dataset_dir
 
 
-def _check_tasks(tasks):
-    for task_ in tasks:
-        bids.validate_label(task_)
-
-
 def _check_entities(entities: dict) -> None:
-    """Validate that all labels are alphanumeric strings."""
-    for value in entities.values():
-        for label_ in value:
+    """Validate that all labels are alphanumeric strings.
+    
+    Parameters
+    ----------
+    entities : :obj:`dict`, optional
+        Extra entity to add to the BIDS filename with a list of values.
+        For example, if you want to add an 'echo' entity
+        with values '1' for some files and '1' for others,
+        you would pass: ``entities={"echo": ['1', '2']}``.
+    """
+    if len(entities.keys()) > 1:
+        # Generating dataset with more than one extra entity
+        # becomes too complex.
+        # Won't be implemented until there is a need.
+        raise ValueError("Only a single extra entity is supported for now.")
+
+    for key in entities:
```

Comment:
Until multiple keys are permitted, the for loop is extraneous.

Reply:
would leave as is so that at least does not have to be changed in case we start supporting several entities